docs/_docs/04_reference/config-blocks-and-attributes.md

```diff
@@ -887,8 +887,17 @@ The `dependency` block supports the following arguments:
 - `mock_outputs_allowed_terraform_commands` (attribute): A list of Terraform commands for which `mock_outputs` are
   allowed. If a command is used where `mock_outputs` is not allowed, and no outputs are available in the target module,
   Terragrunt will throw an error when processing this dependency.
-- `mock_outputs_merge_with_state` (attribute): When `true`, `mock_outputs` and the state outputs will be merged. That is,
-  the `mock_outputs` will be treated as defaults and the real state outputs will overwrite them if the keys clash.
+- `mock_outputs_merge_with_state` (attribute): DEPRECATED. Use `mock_outputs_merge_strategy_with_state`. When `true`, 
+  `mock_outputs` and the state outputs will be merged. That is, the `mock_outputs` will be treated as defaults and the
+  real state outputs will overwrite them if the keys clash.
+- `mock_outputs_merge_strategy_with_state` (attribute): Specifies how any existing state should be merged into the 
+  mocks. Valid values are
+  - `no_merge` (default) - any existing state will be used as is. If the dependency does not have an existing state (it 
+    hasn't been applied yet), then the mocks will be used
+  - `shallow` - the existing state will be shallow merged into the mocks. Mocks will only be used where the output does 
+    not already exist in the dependency's state
+  - `deep` - the existing state will be deeply merged into the mocks. If an output is a map, the mock key will be 
+    used where that key does not exist in the state. Lists will not be merged
```

Comment:
Can you call this `deep_map_only`, to differentiate with `deep` merge that appends the lists?